src/main/java/me/chrr/scribble/mixin/BookEditScreenMixin.java

```diff
@@ -274,50 +315,92 @@ public void init(PlayerEntity player, ItemStack itemStack, Hand hand, CallbackIn
+
+            // ToDo replace with manager.applyModifiersForSelection(activeModifiers) call
+            //  to have the single state of truth for activeModifiers.
+            getRichSelectionManager().toggleModifierForSelection(modifier, toggled);
+        });
+        commandManager.execute(command);
+    }
+
     @Inject(method = "init", at = @At(value = "HEAD"))
     private void initScreen(CallbackInfo ci) {
         initButtons();
-
-        // We need to update the states of all the buttons again.
-        this.getRichSelectionManager().updateSelectionFormatting();
     }
 
     @Inject(method = "updateButtons", at = @At(value = "HEAD"))
     private void updateButtons(CallbackInfo ci) {
-        this.boldButton.visible = !this.signing;
-        this.italicButton.visible = !this.signing;
-        this.underlineButton.visible = !this.signing;
-        this.strikethroughButton.visible = !this.signing;
-        this.obfuscatedButton.visible = !this.signing;
+        Optional.ofNullable(boldButton).ifPresent(button -> button.visible = !this.signing);
+        Optional.ofNullable(italicButton).ifPresent(button -> button.visible = !this.signing);
+        Optional.ofNullable(underlineButton).ifPresent(button -> button.visible = !this.signing);
+        Optional.ofNullable(strikethroughButton).ifPresent(button -> button.visible = !this.signing);
+        Optional.ofNullable(obfuscatedButton).ifPresent(button -> button.visible = !this.signing);
 
         for (ColorSwatchWidget swatch : colorSwatches) {
-            swatch.visible = !this.signing;
+            if (swatch != null) {
+                swatch.visible = !signing;
+            }
         }
 
-        this.deletePageButton.visible = !this.signing && this.richPages.size() > 1;
-        this.insertPageButton.visible = !this.signing;
+        Optional.ofNullable(deletePageButton).ifPresent(button -> button.visible = !signing && richPages.size() > 1);
+        Optional.ofNullable(insertPageButton).ifPresent(button ->
+                button.visible = !signing && richPages.size() < MAX_PAGES_NUMBER
+        );
 
-        this.saveBookButton.visible = !this.signing;
-        this.loadBookButton.visible = !this.signing;
+        Optional.ofNullable(saveBookButton).ifPresent(button -> button.visible = !signing);
+        Optional.ofNullable(loadBookButton).ifPresent(button -> button.visible = !signing);
     }
 
     @Unique
-    private void updateState(@Nullable Formatting color, Set<Formatting> modifiers) {
-        boldButton.toggled = modifiers.contains(Formatting.BOLD);
-        italicButton.toggled = modifiers.contains(Formatting.ITALIC);
-        underlineButton.toggled = modifiers.contains(Formatting.UNDERLINE);
-        strikethroughButton.toggled = modifiers.contains(Formatting.STRIKETHROUGH);
-        obfuscatedButton.toggled = modifiers.contains(Formatting.OBFUSCATED);
-        setSwatchColor(color);
+    private void invalidateFormattingButtons() {
```

Comment:
What's the difference between this method and `updateButtons`? I almost feel like they could be merged now.

Reply:
Overall the logic is the same: read the state from the variable and apply it to the button on the UI. 
`updateButtons ` is still used to update book book control buttons - page editing/changing/creating etc.
`invalidateFormattingButtons ` - for color and modifiers buttons.
From my experience `invalidate` is a more common name for this kind of methods. But it also makes sense to keep the naming consistent. Let me know if I should change it.

Reply:
That makes sense yeah! I think `updateButtons` should be changed to maybe even something like `invalidateButtonVisibility`. I'm not quite sure on the name, but it should be something to differentiate between the two methods.